test/node/bigint.test.ts

```diff
@@ -0,0 +1,102 @@
+import { BSON, BSONError } from '../register-bson';
+import { bufferFromHexArray } from './tools/utils';
+import { expect } from 'chai';
+
+describe('BSON BigInt deserialization support', function () {
+  describe('BSON.deserialize()', function () {
+    type DeserialzationOptions = {
+      useBigInt64: boolean | undefined;
+      promoteValues: boolean | undefined;
+      promoteLongs: boolean | undefined;
+    };
+    type TestTableEntry = {
+      options: DeserialzationOptions;
+      shouldThrow: boolean;
+      expectedResult: BSON.Document;
+      expectedErrorMessage: string;
+    };
+    const testSerializedDoc = bufferFromHexArray(['12', '6100', '2300000000000000']); // key 'a', value 0x23 as int64
+    const useBigInt64Values = [true, false, undefined];
+    const promoteLongsValues = [true, false, undefined];
+    const promoteValuesValues = [true, false, undefined];
+
+    const testTable = useBigInt64Values.flatMap(useBigInt64 => {
+      return promoteLongsValues.flatMap(promoteLongs => {
+        return promoteValuesValues.flatMap(promoteValues => {
+          const useBigInt64Set = useBigInt64 ?? false;
+          const promoteLongsSet = promoteLongs ?? true;
+          const promoteValuesSet = promoteValues ?? true;
```

Comment:
Should probably add comments here explaining that this is done to mirror5 the defaults used in the deserializer

Reply:
Should we let them remain undefined instead? that way we test the defaults set in the src

Reply:
The only place these `xxxSet` flags are used in the test is to determine what our expected result it. We directly pass the value of the option when creating the test. (See lines 41-48)

Reply:
Oh! missed the `Set` I agree with adding a comment, maybe more so to call out that these indicate Set/not Set